src/loader.pl

```diff
@@ -201,8 +201,14 @@
 
 stream_next_term(Stream, Term, LinesRead, Singletons) :-
     (  '$devour_whitespace'(Stream) ->
-       stream_property(Stream, position(position_and_lines_read(_, LinesRead))),
-       read_term(Stream, Term, [singletons(Singletons)])
+       stream_property(Stream, position(position_and_lines_read(Pos0, LinesRead))),
+       (  prolog_load_context(file, File) ->
+          call_with_error_context(
+              read_term(Stream, Term, [singletons(Singletons)]),
+              file-File
+          )
+       ;  read_term(Stream, Term, [singletons(Singletons)])
+       )
```

Comment:
it would appear that another branch bled into here, will need to make a compensating commit